x-pack/plugins/stack_connectors/public/connector_types/pagerduty/links_list.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiButton,
+  EuiButtonIcon,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import {
+  ActionParamsProps,
+  TextFieldWithMessageVariables,
+} from '@kbn/triggers-actions-ui-plugin/public';
+import { PagerDutyActionParams } from '../types';
+
+type LinksListProps = Pick<
+  ActionParamsProps<PagerDutyActionParams>,
+  'index' | 'editAction' | 'errors' | 'messageVariables'
+> &
+  Pick<PagerDutyActionParams, 'links'>;
+
+export const LinksList: React.FC<LinksListProps> = ({
+  editAction,
+  errors,
+  index,
+  links,
+  messageVariables,
+}) => {
+  const areLinksInvalid = Array.isArray(errors.links) && errors.links.length > 0;
+
+  return (
+    <EuiFormRow
+      id="pagerDutyLinks"
+      label={i18n.translate('xpack.stackConnectors.components.pagerDuty.linksFieldLabel', {
+        defaultMessage: 'Links (optional)',
+      })}
+      isInvalid={areLinksInvalid}
+      error={errors.links}
+      fullWidth
+    >
+      <EuiFlexGroup direction="column" data-test-subj="linksList" gutterSize="s">
+        {links &&
+          links.map((link, currentLinkIndex) => (
+            <EuiFlexItem data-test-subj="linksListItemRow">
+              <EuiSpacer size="s" />
+              <EuiFlexGroup key={`my-key-${currentLinkIndex}`}>
+                <EuiFlexItem>
+                  <EuiFormRow
+                    label={i18n.translate(
+                      'xpack.stackConnectors.components.pagerDuty.linkURLFieldLabel',
+                      {
+                        defaultMessage: 'URL',
+                      }
+                    )}
+                    isInvalid={!link.href}
+                  >
+                    <TextFieldWithMessageVariables
+                      index={index}
+                      editAction={(key, value, i) => {
+                        links[currentLinkIndex] = { ...links[currentLinkIndex], href: value };
```

Comment:
I think we should avoid mutating directly the prop. The `editAction` will do it for you.

Reply:
> I think we should avoid mutating directly the prop.

Initially, I had something like

```
const newLinks = [...links]
newLinks[currentLinkIndex] = { ...links[currentLinkIndex], href: value };
editAction('links', newLinks, index)
```

but playing around with it made no difference in the rerenders 😬

Reply:
What do you mean by no difference?

Reply:
I thought changing the prop directly might affect the number of times the component was rerendered but in the end, they were the same. What exactly does it affect?

Reply:
I can see some issues with mutating the props:

1. React is not aware of the change of state
2. We mutate an object that the parent component may use after using the child component
3. We mutate an object that the code in the component assumes is immutable